src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
+  const arrReturn = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'clear') {
+      for (const keyInNewState in newState) {
+        delete newState[keyInNewState];
+      }
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const keyInState in newState) {
+        if (obj.keysToRemove.includes(keyInState)) {
+          delete newState[keyInState];
+        }
+      }
```

Comment:
you can create a cycle for the keysToRemove array. If the property exists in the new state, it will remove. If not - not

```suggestion
      for (const keyToRemove of obj.keysToRemove) {
          delete newState[keyToRemove];
      }
```